From 64487c5877f13f2f37c70df5cfd71e845750f322 Mon Sep 17 00:00:00 2001
From: Evan Huus <eapache@gmail.com>
Date: Fri, 15 Feb 2013 13:10:06 +0000
Subject: [PATCH 05/10] From Alyssa Milburn via
 https://bugs.wireshark.org/bugzilla/show_bug.cgi?id=8335

Make length field unsigned so that negative values fail the bounds check and
throw a regular exception before getting passed to glib (where they cause a
program-ending assert failure instead).

svn path=/trunk/; revision=47672
---
 epan/dissectors/packet-mount.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/epan/dissectors/packet-mount.c b/epan/dissectors/packet-mount.c
index 991be64..c879104 100644
--- a/epan/dissectors/packet-mount.c
+++ b/epan/dissectors/packet-mount.c
@@ -169,7 +169,7 @@ dissect_mount_dirpath_call(tvbuff_t *tvb, int offset, packet_info *pinfo,
 		if(civ->request && (civ->proc==1)){
 			const gchar *host;
 			unsigned char *name;
-			int len;
+			guint32 len;
 			unsigned char *ptr;
 
 			host=ip_to_str(pinfo->dst.data);
-- 
1.7.10.4

